e2e/v3_curl_lease_test.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2018 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+	"testing"
+
+	pb "github.com/coreos/etcd/etcdserver/etcdserverpb"
+)
+
+func TestV3CurlLeaseGrantNoTLS(t *testing.T) {
+	for _, p := range apiPrefix {
+		testCtl(t, testV3CurlLeaseGrant, withApiPrefix(p), withCfg(configNoTLS))
+	}
+}
+func TestV3CurlLeaseRevokeNoTLS(t *testing.T) {
+	for _, p := range apiPrefix {
+		testCtl(t, testV3CurlLeaseRevoke, withApiPrefix(p), withCfg(configNoTLS))
+	}
+}
+func TestV3CurlLeaseLeasesNoTLS(t *testing.T) {
+	for _, p := range apiPrefix {
+		testCtl(t, testV3CurlLeaseLeases, withApiPrefix(p), withCfg(configNoTLS))
+	}
+}
+func TestV3CurlLeaseKeepAliveNoTLS(t *testing.T) {
+	for _, p := range apiPrefix {
+		testCtl(t, testV3CurlLeaseKeepAlive, withApiPrefix(p), withCfg(configNoTLS))
+	}
+}
+
+type v3cURLTest struct {
+	endpoint string
+	value    interface{}
```

Comment:
`value` should `string` type?

Reply:
@gyuho yeah, because of the changing data types I am passing for example pb.LeaseTimeToLiveRequest then pb.LeaseKeepAliveRequest etc I found interface{} useful this generic typing. But perhaps that data flexibility is an issue in itself?

Reply:
Let me revist this.

Reply:
@gyuho you were right I simplified this so that it can handle a string. This is now resolved, thanks!